processing/src/main/java/io/druid/query/MetricsEmittingExecutorService.java

```diff
@@ -51,17 +54,87 @@ protected ListeningExecutorService delegate()
   }
 
   @Override
-  public <T> ListenableFuture<T> submit(Callable<T> tCallable)
+  public <T> ListenableFuture<T> submit(final Callable<T> tCallable)
   {
+    final Callable<T> wrappedPrioritizedCallable = new PrioritizedCallable<T>()
+    {
+      @Override
+      public int getPriority()
+      {
+        if (tCallable instanceof PrioritizedCallable) {
+          return ((PrioritizedCallable) tCallable).getPriority();
+        } else {
+          return 0;
+        }
+      }
+
+      @Override
+      public T call() throws Exception
+      {
+        final Long threadUserTime;
+        if (threadMXBean.isThreadCpuTimeEnabled()) {
+          threadUserTime = threadMXBean.getCurrentThreadUserTime();
+        } else {
+          threadUserTime = null;
+        }
+        final T retval = tCallable.call();
+        if (threadUserTime != null) {
+          emitter.emit(
+              metricBuilder
+                  .setDimension("priority", String.format("%d", getPriority()))
+                  .build("exec/cpu", threadMXBean.getCurrentThreadUserTime() - threadUserTime)
```

Comment:
Currently it reports nanoseconds. On my mac it only has resolution of microseconds.

This is currently used as the general processing pool, but not necessarily only segments. Unless we want the processing pool to be unique to segments I suggest keeping a more general name (but am open to suggestions on the matter)